### DISCUSSION TITLE
Issue: "TypeError: unhashable type: 'dict'" in Poetry 1.8.x and "TypeError: unhashable type: 'Table'" in Poetry 1.7.0

### DISCUSSION THREAD

**Discussion by tsytsyksvitlana** [python-poetry/poetry#9806] [Q&A] [answered]
Issue: "TypeError: unhashable type: 'dict'" in Poetry 1.8.x and "TypeError: unhashable type: 'Table'" in Poetry 1.7.0

I'm encountering different TypeErrors when using Poetry to manage dependencies, even on empty projects. The errors depend on the Poetry version:

With Poetry 1.8.0 or 1.8.4, I get: "TypeError: unhashable type: 'dict'".
With Poetry 1.7.0, I receive: "TypeError: unhashable type: 'Table'".
Here’s what I’ve tried:

Recreated virtual environments using Python 3.13 and 3.12.
Tested on completely fresh projects created through poetry init, yet the errors persist.
Ran tests across various environments: VS Code, PyCharm, and the terminal.
Fully reinstalled Python and Poetry, but the issue remains unresolved.

**Comment by radoering** [comment]
Does `poetry config virtualenvs.path` gives you some output? In this directory you may find an `envs.toml` file which may be corrupt.

Apart from that, your global settings file might be broken. See https://python-poetry.org/docs/configuration/ for the location of the global `config.toml`.

**Reply by tsytsyksvitlana** [reply]
<img width="1227" alt="380814200-2de017ed-25f3-40b8-a822-bf9e9faf738c" src="https://github.com/user-attachments/assets/da7f3900-e587-4823-97d2-9336fa70f284">
Here what I get

**Reply by tsytsyksvitlana** [reply]
```
Last login: Mon Oct 28 21:05:53 on ttys002
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % poetry config --list

cache-dir = "/Users/svitlana_tsytsyk/Library/Caches/pypoetry"
experimental.system-git-client = false
installer.max-workers = null
installer.modern-installation = true
installer.no-binary = null
installer.parallel = true
repositories.pypi.url.url = "https://pypi.org/simple"
virtualenvs.create = true
virtualenvs.in-project = null
virtualenvs.options.always-copy = false
virtualenvs.options.no-pip = false
virtualenvs.options.no-setuptools = false
virtualenvs.options.system-site-packages = false
virtualenvs.path = "{cache-dir}/virtualenvs"  # /Users/svitlana_tsytsyk/Library/Caches/pypoetry/virtualenvs
virtualenvs.prefer-active-python = false
virtualenvs.prompt = "{project_name}-py{python_version}"
warnings.export = true
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % poetry --version

Poetry (version 1.7.0)
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % poetry new your_project_name

Created package your_project_name in your_project_name
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % cd your_project_name  # Navigate into your project directory
poetry add package_name

cd: too many arguments

Poetry could not find a pyproject.toml file in /Users/svitlana_tsytsyk or its parents
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % cd your_project_name                  svitlana_tsytsyk@MacBook-Pro-Svitlana your_project_name % poetry add package_name

unhashable type: 'Table'
svitlana_tsytsyk@MacBook-Pro-Svitlana your_project_name % poetry add black     

unhashable type: 'Table'
svitlana_tsytsyk@MacBook-Pro-Svitlana your_project_name % 
```

**Reply by radoering** [reply]
Ok, so `poetry config virtualenvs.path` does not fail for you. Have you looked into `/Users/svitlana_tsytsyk/Library/Caches/pypoetry` for an `envs.toml` file?

Have you looked into `~/Library/Application Support/pypoetry` for a `config.toml` file?

**Reply by tsytsyksvitlana** [reply]
```
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % ls /Users/svitlana_tsytsyk/Library/Caches/pypoetry/envs.toml

ls: /Users/svitlana_tsytsyk/Library/Caches/pypoetry/envs.toml: No such file or directory
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % cd /Users/svitlana_tsytsyk/Library/Caches/pypoetry          
svitlana_tsytsyk@MacBook-Pro-Svitlana pypoetry % ls
artifacts	cache		virtualenvs
svitlana_tsytsyk@MacBook-Pro-Svitlana pypoetry % ls ~/Library/Application\ Support/pypoetry/config.toml

/Users/svitlana_tsytsyk/Library/Application Support/pypoetry/config.toml
svitlana_tsytsyk@MacBook-Pro-Svitlana pypoetry % cat ~/Library/Application\ Support/pypoetry/config.toml

[repositories.pypi.url]
url = "https://pypi.org/simple"
svitlana_tsytsyk@MacBook-Pro-Svitlana pypoetry % 
```

**Reply by radoering** [reply]
I referenced a wrong path in my previous response. The `envs.toml` is expected in `/Users/svitlana_tsytsyk/Library/Caches/pypoetry/virtualenvs` not in `/Users/svitlana_tsytsyk/Library/Caches/pypoetry`.

**Reply by tsytsyksvitlana** [reply]
```
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % cd /Users/svitlana_tsytsyk/Library/Caches/pypoetry/virtualenvs
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % ls
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % 
```

**Reply by radoering** [reply, marked as answer]
So no `envs.toml`. The content of the `config.toml` does not look bad, but you can try to delete/rename the `config.toml` and try again. If you still get an error, you can run the command that raises the error with `-vvv` and provide the output.

**Reply by tsytsyksvitlana** [reply]
thank you so much!

Last login: Tue Oct 29 18:06:49 on ttys005
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % cd /Users/svitlana_tsytsyk/Library/Caches/pypoetry/virtualenvs
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % ls
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % ls ~/Library/Application\ Support/pypoetry/config.toml
/Users/svitlana_tsytsyk/Library/Application Support/pypoetry/config.toml
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % cat ~/Library/Application\ Support/pypoetry/config.toml
[repositories.pypi.url]
url = "https://pypi.org/simple"
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % mv ~/Library/Application\ Support/pypoetry/config.toml ~/Library/Application\ Support/pypoetry/config_backup.toml

svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % poetry install black

Poetry could not find a pyproject.toml file in /Users/svitlana_tsytsyk/Library/Caches/pypoetry/virtualenvs or its parents
svitlana_tsytsyk@MacBook-Pro-Svitlana virtualenvs % cd ~
svitlana_tsytsyk@MacBook-Pro-Svitlana ~ % mkdir poetry_test_project
cd poetry_test_project

svitlana_tsytsyk@MacBook-Pro-Svitlana poetry_test_project % poetry init

This command will guide you through creating your pyproject.toml config.

Package name [poetry_test_project]:  
Version [0.1.0]:  
Description []:  
Author [Svitlana Tsytsyk <TSYTSYK.svetlana@gmail.com>, n to skip]:  
License []:  
Compatible Python versions [^3.13]:  

Would you like to define your main dependencies interactively? (yes/no) [yes] no
Would you like to define your development dependencies interactively? (yes/no) [yes] no
Generated file

[tool.poetry]
name = "poetry-test-project"
version = "0.1.0"
description = ""
authors = ["Svitlana Tsytsyk <TSYTSYK.svetlana@gmail.com>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.13"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"


Do you confirm generation? (yes/no) [yes] yes
svitlana_tsytsyk@MacBook-Pro-Svitlana poetry_test_project % poetry add requests

Creating virtualenv poetry-test-project-lLD20GiF-py3.13 in /Users/svitlana_tsytsyk/Library/Caches/pypoetry/virtualenvs
Using version ^2.32.3 for requests

Updating dependencies
Resolving dependencies... (1.2s)

Package operations: 5 installs, 0 updates, 0 removals

  • Installing certifi (2024.8.30)
  • Installing charset-normalizer (3.4.0)
  • Installing idna (3.10)
  • Installing urllib3 (2.2.3)
  • Installing requests (2.32.3)

Writing lock file
svitlana_tsytsyk@MacBook-Pro-Svitlana poetry_test_project % poetry add black
Using version ^24.10.0 for black

Updating dependencies
Resolving dependencies... (1.7s)

Package operations: 6 installs, 0 updates, 0 removals

  • Installing click (8.1.7)
  • Installing mypy-extensions (1.0.0)
  • Installing packaging (24.1)
  • Installing pathspec (0.12.1)
  • Installing platformdirs (4.3.6)
  • Installing black (24.10.0)

Writing lock file
svitlana_tsytsyk@MacBook-Pro-Svitlana poetry_test_project %
